docs/specifications/linking-and-embedding-specification.md

```diff
@@ -0,0 +1,364 @@
+- **Edit-time**: Activities that happen while a markdown file is edited, are said to be edit-time activities.
+- **EBlock**: Embeddable block; Piece of text in a note that can be embedded. See block embedding.
+- **Frontmatter**: Metadata section at the beginning of a note.
+  *See Also*: https://jekyllrb.com/docs/front-matter/
+- **Graph**: Graph structure that contains information about all the files in the workspace. Notes in the workspace are analyzed further e.g. for links and link targets.
+  *Also known as*: Index.
+- **Id**: Element attribute in the generated HTML that can be targeted with links / block embedding. Ids can also be used within the markdown files when defining anchors.
+- **Indexing**: Operation that goes through all the markdown files in the Foam workspace, finding all the links, block embeds, and link targets and finally linking these together in a Graph.
+- **Link Reference Definitions**: Information in a markdown file that converts wikilinks into regular markdown links.
+  *Also known as*: LinkRefDefs.
+  *See Also*: https://github.github.com/gfm/#link-reference-definitions
+- **Link**: A piece of text that points to a Target.
+- **Note**: A file containing text in markdown format.
+  *Also known as*: Foam Bubble, Document, Markdown file.
+- **Tag**: A piece of text that either acts as a link, or as an abstract categorization.
+- **Target**: An item (e.g. note, heading or an anchor) that can be pointed to with a link.
+- **Title**: The H1 heading of a note. A note can contain only a single title.
+- **Workspace**: The location (folder) where Foam Notes live in.
+
+## Targets
+
+The following items can be targetted via linking and / or embedding. The separation is explained in more detail in [Linking](#linking) and [Embedding](#embedding) chapters.
+
+### Notes
+
+Notes in Foam are individual markdown files. Notes can both reside in the Foam workspace root and any of its sub-folders. Both files and folders are encouraged to follow the kebab-case naming convention (e.g. `some/path-to/a-file.md`).
+
+### Headings
+
+Headings in Foam are denoted with markdown heading syntax, e.g. `## Heading on Second Level`. Each heading will have a corresponding id, a kebab-case version of the text (e.g. `heading-on-second-level`). In case there's multiple headings with the same text, their id is suffixed with a running number (e.g. `my-heading`, `my-heading-1`, `my-heading-2`).
+
+### Anchors
+
+User can add arbitrary anchors into a note with HTML-syntax: `<a name="my-anchor"></a>`. Since these names (identifiers) are written by users, they might end up being non-unique within a note. In these cases, an error is shown to the user. See the reasoning of using the name-attribute instead of the id-attribute from https://stackoverflow.com/a/7335259. In the future, better syntax for the anchors might be developed.
```

Comment:
Proposal: Use auto-closing anchor tags by default:

```suggestion
User can add arbitrary anchors into a note with HTML-syntax: `<a name="my-anchor" />`. Since these names (identifiers) are written by users, they might end up being non-unique within a note. In these cases, an error is shown to the user. See the reasoning of using the name-attribute instead of the id-attribute from https://stackoverflow.com/a/7335259. In the future, better syntax for the anchors might be developed.
```

I know technically this is not valid HTML, but it does work:
https://gist.github.com/jevakallio/086e3b375af45a82cc46d0ddd2f4fb39

IMO using self-closing tags in documentation makes this a lot more palatable to use. Folks who prefer their HTML valid could have that option.